components/director/internal/domain/apptemplate/resolver.go

```diff
@@ -246,13 +251,30 @@ func (r *Resolver) CreateApplicationTemplate(ctx context.Context, in graphql.App
 
 	selfRegID := r.uidService.Generate()
 	convertedIn.ID = &selfRegID
-	validate := func() error {
-		return validateAppTemplateForSelfReg(in.ApplicationInput)
-	}
 
-	selfRegLabels, err := r.selfRegManager.PrepareForSelfRegistration(ctx, resource.ApplicationTemplate, convertedIn.Labels, selfRegID, validate)
+	consumerInfo, err := consumer.LoadFromContext(ctx)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrapf(err, "while loading consumer")
+	}
+
+	labels := convertedIn.Labels
+	if _, err := tenant.LoadFromContext(ctx); err == nil && consumerInfo.Flow.IsCertFlow() {
+		isSelfReg, selfRegFlowErr := r.isSelfRegFlow(labels)
+		if selfRegFlowErr != nil {
+			return nil, selfRegFlowErr
+		}
+
+		if isSelfReg {
+			labels[scenarioassignment.SubaccountIDKey] = consumerInfo.ConsumerID
```

Comment:
I think we want this label when we are not only in self reg flow, but also in cld flow.